hooks/useSharedGuideIds.tsx

```diff
@@ -0,0 +1,58 @@
+import {
+  useState,
+  useEffect,
+  useContext,
+  createContext,
+  ReactNode,
+  Dispatch,
+  SetStateAction,
+} from "react";
+import { SHARED_GUIDES_LOCAL_STORAGE } from "utils/constants";
+
+const SHARED_GUIDES_INITIAL_STATE = [""];
+
+interface SharedGuideIdsContextResult {
+  sharedGuideIds: string[];
+  setSharedGuideIds: Dispatch<SetStateAction<string[]>>;
+}
+
+const SharedGuideIdsContext = createContext<SharedGuideIdsContextResult>(
+  {} as SharedGuideIdsContextResult
+);
+
+function getGuideIds() {
+  const rawSharedGuideIds = localStorage.getItem(SHARED_GUIDES_LOCAL_STORAGE);
+
+  const sharedGuideIds =
+    rawSharedGuideIds?.split("|") || SHARED_GUIDES_INITIAL_STATE;
```

Comment:
I think a better way to handle saving to local storage is using `JSON.stringify` and `JSON.parse` 

That way we have a nice object to work with (esp if we need to add more data) and don't have to mess with strings

Reply:
agreed, will help with sad paths / error handling as well